File: src/yanzm/example/adaptersamples/SimpleAdapterSample2.java
package yanzm.example.adaptersamples;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import android.app.ListActivity;
import android.os.Bundle;
import android.widget.SimpleAdapter;

public class SimpleAdapterSample2 extends ListActivity {
	
    @SuppressWarnings({ "unchecked", "serial" })
	public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final List<HashMap<String, Integer>> data = 
            Arrays.asList(
                new HashMap<String, Integer>() {{
                    put("Icon1", R.drawable.alermclock);  
                    put("Icon2", R.drawable.bookopen);
                }},
                new HashMap<String, Integer>() {{
                    put("Icon1", R.drawable.calculator);  
                    put("Icon2", R.drawable.camera);
                }}
            );

        String[] from = {"Icon1", "Icon2"};
        
        int[] to = {R.id.imageview1, R.id.imageview2};
        
        SimpleAdapter simpleAdapter = new SimpleAdapter(this, data, R.layout.list_item2, from, to);
        setListAdapter(simpleAdapter);
    }
}

File: app/src/main/java/yanzm/example/adaptersamples/SimpleAdapterSample2.kt
package yanzm.example.adaptersamples

import android.os.Bundle
import android.widget.ListView
import android.widget.SimpleAdapter
import androidx.appcompat.app.AppCompatActivity

class SimpleAdapterSample2 : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listView = ListView(this)
        setContentView(listView)

        val data = listOf(
            mapOf(
                "Icon1" to R.drawable.ic_alarm,
                "Icon2" to R.drawable.ic_book
            ),
            mapOf(
                "Icon1" to R.drawable.ic_calculator,
                "Icon2" to R.drawable.ic_camera
            ),
        )

        val from = arrayOf("Icon1", "Icon2")
        val to = intArrayOf(R.id.imageview1, R.id.imageview2)

        listView.adapter = SimpleAdapter(this, data, R.layout.list_item2, from, to)
    }
}
